service/policy/db/utils.go

```diff
@@ -27,9 +30,35 @@ var createSuffix = "RETURNING id, " + constructMetadata("", false)
 func unmarshalMetadata(metadataJSON []byte, m *common.Metadata, logger *logger.Logger) error {
 	if metadataJSON != nil {
 		if err := protojson.Unmarshal(metadataJSON, m); err != nil {
-			logger.Error("could not unmarshal metadata", slog.String("error", err.Error()))
+			logger.Error("failed to unmarshal metadata", slog.String("error", err.Error()), slog.String("metadataJSON", string(metadataJSON)))
```

Comment:
Up the chain, `db.StatusifyError` will log any error that occurred throughout policy CRUD before responding to the client request, so it may be better here to wrap these bits in a format string and return it instead of logging multiple times for the same issue.
```suggestion
return fmt.Errorf("failed to unmarshal metadataJSON [%s]: %w", string(metadataJSON), err)
```

Same comment below with the `unmarshalAttributeValue`.

Reply:
I like that idea and in general I think we should always just return errors and let calls further up the chain handle logging.  Also has the benefit of not having to pass the pointer to the logger instance around everywhere.